pontoon/base/models.py

```diff
@@ -2585,6 +2585,29 @@ def reset_active_translation(self, locale, plural_form=None):
         else:
             return Translation()
 
+    def reset_term_translation(self, locale):
```

Comment:
I noticed the loop in the bulk edit, which made me wonder if it's worth to refactor this to have a query of Entities?

Only if this would be a new bottleneck, I guess?